### ISSUE TITLE
Memory leak in ABEND

### ISSUE THREAD

**Issue by yoavnir** [yoavnir/specs2016#85]
This is test 109 in `ProcessingTest`:

```
==812== Memcheck, a memory error detector
==812== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==812== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==812== Command: ../exe/ProcessingTest 109
==812==
Test #109 ***** OK *****: <ABEND: too big>

All tests passed.
==812==
==812== HEAP SUMMARY:
==812==     in use at exit: 49 bytes in 2 blocks
==812==   total heap usage: 706 allocs, 704 frees, 43,168 bytes allocated
==812==
==812== 49 (16 direct, 33 indirect) bytes in 1 blocks are definitely lost in loss record 2 of 2
==812==    at 0x4C2A1E3: operator new(unsigned long) (vg_replace_malloc.c:334)
==812==    by 0x43C948: SpecString::newString() (SpecString.cc:85)
==812==    by 0x438C54: StringBuilder::insert(SpecString*, unsigned long, bool) (StringBuilder.cc:122)
==812==    by 0x42800E: DataField::apply(ProcessingState&, StringBuilder*) (dataField.cc:504)
==812==    by 0x420C7A: itemGroup::processDo(StringBuilder&, ProcessingState&, Reader*) (specItems.cc:247)
==812==    by 0x4131C2: runTestOnExample(char const*, char const*) (ProcessingTest.cc:88)
==812==    by 0x407329: runTestOnExample (ProcessingTest.cc:139)
==812==    by 0x407329: main (ProcessingTest.cc:432)
==812==
==812== LEAK SUMMARY:
==812==    definitely lost: 16 bytes in 1 blocks
==812==    indirectly lost: 33 bytes in 1 blocks
==812==      possibly lost: 0 bytes in 0 blocks
==812==    still reachable: 0 bytes in 0 blocks
==812==         suppressed: 0 bytes in 0 blocks
==812==
==812== For counts of detected and suppressed errors, rerun with: -v
==812== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
```

**Comment by yoavnir** [comment]
Probably a duplicate of one of the leaks in #84

**Comment by yoavnir** [comment]
Note that this does not reproduce on Linux with newer GCC.

Will need to reproduce after fix to #84 on older Linux.

**Comment by yoavnir** [comment]
No longer reproduces on either Linux type
